### ISSUE TITLE
[Serious] State: Table sort state is missing or incorrect

### ISSUE THREAD

**Issue by ntallman** [APTrust/registry#103]
## Issue Details

| | |
|---|---|
| **Source Issue ID** | 87582 |
| **Checkpoint** | 4.1 Compatible › Name, Role, Value (4.1.2.a) |
| **Impact** | Serious |
| **Standards** | wcag2a, wcag2aa, wcag21a, wcag21aa, wcag22a, wcag22aa |
| **Test Area** | 05 Add New User Workflow |
| **Test Case Name** | AP Trust Registry (Demo) |
| **URL** | <https://demo.aptrust.org/users> |
| **Environment** | Production |
| **Platform** | Windows | Chrome | Firefox |
| **Assistive Technology** | JAWS | NVDA | Magnifier |
| **Method** | Manual |
| **Digital Asset Type** | Desktop Web |
| **Release** | 4.2026 |
| **Product** | Website |
| **Date Reported** | 05/06/26 at 12:10 pm |
| **Unit Type** | Page |
| **Issue Type** | Accessibility |
| **Reported By** | a3c.access@gmail.com |

## Description

The sorted state (ascending/descending) of table columns is missing or incorrect.
Pattern: Table columns that can be sorted do not announce as sortable or convey their sorted status. In the Users table, the Name, Email, Role, Institution, Last Sign In, and Disabled columns can be sorted, but it does not indicate that these columns are sortable or their current state (e.g. ascending, descending). See Managing Users User Experience Video at 1:30. 
Also occurs on 03 Objects page and 04 Work Items

## Recommendation

Rule : 
The name, role, value, states, and properties of user interface components MUST be programmatically determinable by assistive technologies.

How To Fix: 
Fix this issue by using ONE of the following techniques:
1. Use the aria-sort attribute values of "ascending" or "descending" or "none" on the column header to indicate the sort state of the data in a sortable column. This technique also requires that role="columnheader" be added to the <th> element.

<th role="columnheader" aria-sort="descending">
<button class="sortableColumnLabel">Name</button>
</th>

2. Add CSS clipped (visually hidden) text to the sorting element to indicate the sorted state.

<th>
<div tabindex="0" role="button" id="date-sort">
Date <span class="sr-text">sorted ascending</span>
</div>
</th>

Reference: 
WAI-ARIA Recommendation: https://www.w3.org/TR/wai-aria-1.1/#aria-sort

Background: 
States and properties are attributes used to convey essential information about an element to screen readers and other assistive technologies. Some roles require certain state and property information - such as the checked/unchecked state of a checkbox. Native HTML elements provide those required states and properties, so nothing more needs to be done. If you create a custom version of a native HTML element or a custom control or widget that does not have a native HTML equivalent, you must add the relevant states and properties using ARIA.

## Source Code

```
<th>
          <a href="/users?sort=email__desc" class="is-grey-dark">Email<span class="material-icons sort-icon" aria-hidden="true">
              keyboard_arrow_up</span></a>
        </th>
```

## Screenshots

![05_table sort](https://raw.githubusercontent.com/APTrust/registry/master/docs/accessibility-audit/screenshots/87582/05_table_sort.png)
